src/Hogwarts.java

```diff
@@ -0,0 +1,80 @@
+public abstract class Hogwarts {
+    private String name;
+    private int powerOfMagic;
+    private int transgression;
+
+    public Hogwarts(String  name, int powerOfMagic, int transgression) {
+        this.name=name;
+        this.powerOfMagic = powerOfMagic;
+        this.transgression = transgression;
+    }
+
+    abstract public int calculateSpecificScore();
+    public void compare (Hogwarts hogwarts) {
+        if (this.getClass().equals(hogwarts.getClass())) {
+            compareSpecificScore(hogwarts);
+        } else {compareGeneralScore(hogwarts);
+        }
```

Comment:
```suggestion
        compareGeneralScore(hogwarts);
```
не нужно делать никакую проверку класса, надо просто вызвать метод сравнения по базовым характеристикам